src/nnet3/nnet-computation.cc

```diff
@@ -413,6 +608,210 @@ void NnetComputation::Print(std::ostream &os, const Nnet &nnet) const {
+  WriteToken(os, binary, "<MatrixDebugInfo>");
+  for (size_t c = 0; c < matrix_debug_info.size(); c++) {
+    matrix_debug_info[c].Write(os, binary);
+  }
+
+  WriteToken(os, binary, "<NumSubMatrices>");
+  WriteBasicType(os, binary, submatrices.size());
+  WriteToken(os, binary, "<SubMatrices>");
+  for (size_t c = 0; c < submatrices.size(); c++) {
+    submatrices[c].Write(os, binary);
+  }
+
+//  WriteToken(os, binary, "<NumComponentPrecomputedIndexes>");
+//  WriteBasicType(os, binary, component_precomputed_indexes.size());
+//  WriteToken(os, binary, "<ComponentPrecomputedIndexes>");
+//  for (int32 c = 0; c < component_precomputed_indexes.size(); c++) {
+//    if (component_precomputed_indexes[c] != NULL) 
+//      component_precomputed_indexes[c]->Write(os, binary);
+//  }
+
+  WriteToken(os, binary, "<NumIndexes>");
+  WriteBasicType(os, binary, indexes.size());
+  WriteToken(os, binary, "<Indexes>");
+  for (size_t c = 0; c < indexes.size(); c++) {
+    WriteBasicType(os, binary, indexes[c].size());
+    for (size_t i = 0; i < indexes[c].size(); i++)
+      WriteBasicType(os, binary, indexes[c][i]);
+  }
+
+  WriteToken(os, binary, "<NumIndexesMulti>");
+  WriteBasicType(os, binary, indexes_multi.size());
+  WriteToken(os, binary, "<IndexesMulti>");
+  for (size_t c = 0; c < indexes_multi.size(); c++) {
+    WriteBasicType(os, binary, indexes_multi[c].size());
+    for (size_t i = 0; i < indexes_multi[c].size(); i++) {
+      WriteBasicType(os, binary, indexes_multi[c][i].first);
+      WriteBasicType(os, binary, indexes_multi[c][i].second);
+    }
+  }
+
+  WriteToken(os, binary, "<NumIndexesRanges>");
+  WriteBasicType(os, binary, indexes_ranges.size());
+  WriteToken(os, binary, "<IndexesRanges>");
+  for (size_t c = 0; c < indexes_ranges.size(); c++) {
+    WriteBasicType(os, binary, indexes_ranges[c].size());
+    for (size_t i = 0; i < indexes_ranges[c].size(); i++) {
+      WriteBasicType(os, binary, indexes_ranges[c][i].first);
+      WriteBasicType(os, binary, indexes_ranges[c][i].second);
+    }
+  }
+
+  WriteToken(os, binary, "<NumInputOutputInfo>");
+  WriteBasicType(os, binary, input_output_info.size());
+  WriteToken(os, binary, "<InputOutputInfo>");
+  std::map<int32, std::pair<int32, int32> > input_output_info_cp(input_output_info.begin(), input_output_info.end());
+  for (std::map<int32, std::pair<int32, int32> >::const_iterator iter =
+           input_output_info_cp.begin(); iter != input_output_info_cp.end(); ++iter) {
+    WriteBasicType(os, binary, iter->first);
+    WriteBasicType(os, binary, iter->second.first);
+    WriteBasicType(os, binary, iter->second.second);
+  }
+
+  WriteToken(os, binary, "<NumCommands>");
+  WriteBasicType(os, binary, commands.size());
+  WriteToken(os, binary, "<Commands>");
+  for (size_t c = 0; c < commands.size(); c++) {
+    commands[c].Write(os, binary);
+  }
+
+  WriteToken(os, binary, "<NeedModelDerivative>");
+  WriteBasicType(os, binary, need_model_derivative);
+}
```

Comment:
Don't forget an opening and closing tag <NnetComputation> and </NnetComputation>